fastify.js

```diff
@@ -794,16 +794,12 @@ function fastify (options) {
       // only call isAsyncConstraint once
       if (isAsync === undefined) isAsync = router.isAsyncConstraint()
       if (rewriteUrl) {
-        const originalUrl = req.url
-        const url = rewriteUrl(req)
-        if (originalUrl !== url) {
-          logger.debug({ originalUrl, url }, 'rewrite url')
-          if (typeof url === 'string') {
-            req.url = url
-          } else {
-            const err = new FST_ERR_ROUTE_REWRITE_NOT_STR(req.url, typeof url)
-            req.destroy(err)
-          }
+        const url = rewriteUrl.call({ log: logger }, req)
```

Comment:
Another question: why not just let the user use `req.log` directly?

Reply:
Sorry, @fox1t, my edit probably changes what you were agreeing with.

Reply:
Oh. Yes. I still prefer having this bound to instance/req, since it is something that a Fastify user could expect.

Reply:
@mcollina and @Eomm do you agree that `rewriteUrl(req, { logger })` is a non-breaking implementation?

Reply:
> Another question: why not just let the user use `req.log` directly?

@fox1t At this point in the code, `req` is the raw HTTP message, not the Fastify request object. As far as I can tell the Fastify request object is created post-routing. I agree it would be great to have any logs bound to the request context, but I think it would require some significant refactoring...

Reply:
This would be amazing. @Eomm and @jsumners thoughts?

Reply:
I think that is acceptable. However, the documentation needs to be clear that the logger is not yet attached to the request (this needs to be clear in the docs regardless of direction).

Reply:
Agree - it matches the fastify pattern too

Reply:
Ok, done.

> I think that is acceptable. However, the documentaton needs to be clear that the logger is not yet attached to the request (this needs to be clear in the docs regardless of direction).

The docs already aluded to this but I made it a bit more explicit regarding the fact that `this` is the root fastify object and `req` is not the FastifyRequest object. What do you think?